### PR TITLE
Invoke automated redemptions periodically by maintainer

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Refs: https://github.com/keep-network/keep-core/issues/3614
Depends on: https://github.com/keep-network/keep-core/pull/3659

Here we add automation for the redemption proposal submission logic. Every 3 hours, the maintainer will check for pending redemptions starting from the oldest wallet and will propose them for redemption by submitting a proposal to the `WalletCoordinator` contract.

### CODE REVIEW

File: pkg/maintainer/wallet/redemptions.go [hunk 1]
```diff
@@ -11,6 +12,43 @@ import (
 	"time"
 )
 
+func (wm *walletMaintainer) runRedemptionTask(ctx context.Context) error {
+	redemptionMaxSize, err := wm.chain.GetRedemptionMaxSize()
+	if err != nil {
+		return fmt.Errorf("failed to get deposit sweep max size: [%w]", err)
```
Comment (pdyraga) [inline pkg/maintainer/wallet/redemptions.go:18]:
```suggestion
		return fmt.Errorf("failed to get redemption max size: [%w]", err)
```

File: pkg/maintainer/wallet/redemptions.go [hunk 2]
```diff
@@ -11,6 +12,43 @@ import (
...
+	}
+
+	walletPublicKeyHash, redeemersOutputScripts, err := FindPendingRedemptions(
+		wm.chain,
+		[20]byte{},
+		redemptionMaxSize,
+	)
```
Comment (pdyraga) [inline pkg/maintainer/wallet/redemptions.go:25]: `FindPendingRedemptions` in the current form stops searching when it finds a wallet with a pending redemption. I don't think this is enough. A malicious redeemer can easily make redemption requests to other wallets time out. I think we should - at a minimum - introduce a parameter that will limit the number of wallets that can process redemptions at the same time and not stop searching in `FindPendingRedemptions` until we reach the limit.

Maybe it's already on your radar for this PR, just making sure.

Comment (lukasz-zimnoch) [reply pkg/maintainer/wallet/redemptions.go:25]: Yep, this is captured in the task list of https://github.com/keep-network/keep-core/issues/3614

File: pkg/maintainer/wallet/redemptions.go (hunk 1 above)
Comment (lukasz-zimnoch) [reply pkg/maintainer/wallet/redemptions.go]: See 14bca05a

**Review by pdyraga — APPROVED** [review]

**Pull request merged**